rust-wasm-id-allocator/wasm-id-allocator/src/lib.rs

```diff
@@ -205,3 +201,225 @@ impl InteropIdRange {
         self.count
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use distributed_id_allocator::compressor::{NormalizationError, SessionTokenError};
+
+    static SESSION_ID_1: &str = "748540ca-b7c5-4c99-83ff-c1b8e02c09d6";
+
+    #[test]
+    #[should_panic]
+    fn cluster_capacity_fract() {
+        let mut compressor = IdCompressor::new(String::from(SESSION_ID_1)).unwrap();
+        _ = compressor.set_cluster_capacity(5.5 as f64);
+    }
+
+    #[test]
+    #[should_panic]
+    fn cluster_capacity_negative() {
+        let mut compressor = IdCompressor::new(String::from(SESSION_ID_1)).unwrap();
+
+        _ = compressor.set_cluster_capacity(-2 as f64);
+    }
+
+    #[test]
+    fn generate_next_id() {
+        let mut compressor = IdCompressor::new(String::from(SESSION_ID_1)).unwrap();
+
+        assert_eq!(compressor.generate_next_id(), -1 as f64);
+    }
+
+    #[test]
+    #[should_panic]
+    fn get_token_invalid_uuid() {
+        let mut compressor = IdCompressor::new(String::from(SESSION_ID_1)).unwrap();
+
+        _ = compressor.get_token(String::from("not_a_uuid")); // Errors at SessionId::from_uuid_string()
```

Comment:
Why does this panic? I thought it should return a result with a JsError.

Reply:
JsError generation fails with a panic stating that JsError is not available unless building to the wasm32 target. My guess is that receiving and checking the JsErrors in the TS layer will work fine, but they do not work here in Rust.